src/right-field/results-table/ResultsTable.jsx

```diff
@@ -16,45 +16,52 @@ if(config.queryFolder.substring(config.queryFolder.length-1) !== '/'){
   config.queryFolder = `${config.queryFolder}/`
 }
 
-const QueryEngine = require('@comunica/query-sparql').QueryEngine;
-const myEngine = new QueryEngine()
-
+let queryWorker = undefined
 
 /**
  * 
- * @param {query} props.query The query (as defined in the config file) that should be executed and results displayed in the table. 
+ * @param {query} props.query The query undefined(as defined in the config file) that should be executed and results displayed in the table. 
+ * @param {query} props.query The query undefined(as defined in the config file) that should be executed and results displayed in the table. 
  * @returns {Component} A React component giving a structural representation of the query results.  
  */
 function ResultsTable(props){
     const selectedquery = props.selectedquery
     const [results, setResults] = useState([])
     const [variables, setVariables] = useState([])
     const containerRef = useRef()
+    const [isQuerying, setQuerying] = useState(false)
 
     if(containerRef.current){
-      console.log(containerRef.current.wrapper.current)
       containerRef.current.wrapper.current.className = "grid-wrapper";
     }  
 
     let adder = (item, variables) => setResults((old) => {
       let newValues = []
       for(let variable of variables){
-        let value = item.get(variable) ? item.get(variable) : ""
+        let value = item[variable] ? item[variable] : ""
         let type = variable.split('_')[1]
         let componentCaller = typeRepresentationMapper[type] 
-        componentCaller = componentCaller ? componentCaller : (text) => text.id
+        componentCaller = componentCaller ? componentCaller : (text) => text.value
         newValues.push(componentCaller(value))
       }
       
       return [...old, newValues]}
       
     )
+    
+    useEffect(() => {configureQueryWorker(adder, setVariables, setQuerying)}, [])
+    
 
     let onqueryChanged = () => {
```

Comment:
Why is this variable declared with `let`?

Reply:
Probably something I forget to change, this is now adjusted respectively in [commit 4b42c9a](https://github.com/SolidLabResearch/generic-data-viewer/pull/14/commits/4b42c9a4db57771ec72ec43023cb3daac5879280)